tests/common.go

```diff
@@ -291,7 +413,163 @@ func ValidateVolumes(ctx *scheduler.Context, errChan ...*chan error) {
+		})
+
+		var vols []*volume.Volume
+		Step(fmt.Sprintf("get %s app's pure volumes", ctx.App.Key), func() {
+			vols, err = Inst().S.GetPureVolumes(ctx)
+			processError(err, errChan...)
+		})
+
+		for _, vol:= range vols {
+
+			Step(fmt.Sprintf("get %s app's volume: %s then create snapshot using pxctl", ctx.App.Key, vol), func() {
+				err = Inst().V.ValidateCreateSnapshotUsingPxctl(vol.ID)
+				if err != nil {
+					expect(err.Error()).To(contain(errPureFileSnapshotNotSupported.Error()))
+				}
+			})
+			Step(fmt.Sprintf("get %s app's volume: %s then create cloudsnap using pxctl", ctx.App.Key, vol), func() {
+				err = Inst().V.ValidateCreateCloudsnapUsingPxctl(vol.ID)
+				if err != nil {
+					expect(err.Error()).To(contain(errPureFileSnapshotNotSupported.Error()))
+				}
+			})
+		}
+		Step(fmt.Sprintf("validating groupsnap for using pxctl"), func() {
+			err = Inst().V.ValidateCreateGroupSnapshotUsingPxctl()
+			if err != nil {
+				expect(err.Error()).To(contain(errPureFileSnapshotNotSupported.Error()))
+			}
+		})
+	})
+}
+
+// ValidateResizeFBPVC is the ginkgo spec for validating resize of volumes
+func ValidateResizeFBPVC(ctx *scheduler.Context, errChan ...*chan error) {
+	context("For validation of an resizing pvc", func() {
+		var err error
+		Step(fmt.Sprintf("inspect %s app's volumes", ctx.App.Key), func() {
+			appScaleFactor := time.Duration(Inst().GlobalScaleFactor)
+			err = Inst().S.ValidateVolumes(ctx, appScaleFactor*defaultTimeout, defaultRetryInterval, nil)
+			processError(err, errChan...)
+		})
+
+		Step(fmt.Sprintf("validating resizing pvcs"), func() {
+			err = Inst().S.ResizePureVolumes(ctx)
+			if err != nil {
+				expect(err).ToNot(haveOccurred())
+			}
+		})
+	})
+}
+
+// ValidateVolumeStatsticsDynamicUpdate is the ginkgo spec for validating dynamic update of byteUsed statstic for pure volumes
+func ValidateVolumeStatsticsDynamicUpdate(ctx *scheduler.Context, errChan ...*chan error) {
+	context("For validation of an resizing pvc", func() {
+		var err error
+		Step(fmt.Sprintf("inspect %s app's volumes", ctx.App.Key), func() {
+			appScaleFactor := time.Duration(Inst().GlobalScaleFactor)
+			err = Inst().S.ValidateVolumes(ctx, appScaleFactor*defaultTimeout, defaultRetryInterval, nil)
+			processError(err, errChan...)
+		})
+		var vols []*volume.Volume
+		Step(fmt.Sprintf("get %s app's pure volumes", ctx.App.Key), func() {
+			vols, err = Inst().S.GetPureVolumes(ctx)
+			processError(err, errChan...)
+		})
+		byteUsedInitial, err := Inst().V.ValidateGetByteUsedForVolume(vols[0].ID, make(map[string]string))
+		fmt.Println(fmt.Sprintf("initially the byteUsed is %v", byteUsedInitial))
+		// get the pod for this pvc
+		pods, err := Inst().S.GetPodsForPVC(vols[0].Name, vols[0].Namespace)
+
+		// write to the FB volume
+		cmdArgs := []string{"exec", "-it", pods[0].Name, "-n", pods[0].Namespace, "--", "bash","-c", "dd bs=521 count=1048576 </dev/urandom > "+ getMountPath(pods[0].Namespace)+ "/myfile"}
```

Comment:
Could we maybe split this out across multiple lines? It's quite an eyesore to read right now

Reply:
`bs` should be 512, not 521